src/Database/Esqueleto/Internal/Internal.hs

```diff
@@ -531,15 +651,24 @@ val v = ERaw Never $ const ("?", [toPersistValue v])
 
 -- | @IS NULL@ comparison.
 isNothing :: PersistField typ => SqlExpr (Value (Maybe typ)) -> SqlExpr (Value Bool)
-isNothing (ERaw p f)        = ERaw Parens $ first ((<> " IS NULL") . parensM p) . f
 isNothing (ECompositeKey f) = ERaw Parens $ flip (,) [] . (intersperseB " AND " . map (<> " IS NULL")) . f
+isNothing v = ERaw Parens $ first ((<> " IS NULL")) . g
+  where 
+    g =
+      case v of 
+        ERaw p f             -> first (parensM p) . f
+        EAliasedValue i _    -> aliasedValueIdentToRawSql i
+        EValueReference i i' -> valueReferenceToRawSql i i'
+        ECompositeKey _      -> undefined -- defined above
```

Comment:
Can you restructure this code so that we don't have an unsafe pattern match here?